### PR TITLE
Add `awsbase.ErrCodeEquals`, AWS SDK for Go v2 variant of helper in `v2/awsv1shim/tfawserr`

### PR DESCRIPTION
**Pull request by ewbankkit**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Adds an AWS SDK for Go v2 variant of the `ErrCodeEquals` helper in `v2/awsv1shim/tfawserr`.
Not all services in the AWS SDK for Go v2 uses [strongly-typed, modeled errors](https://aws.github.io/aws-sdk-go-v2/docs/handling-errors/#api-error-responses) (EC2 is top of mind).
To support testing the returned API error code for these servies, add a variant of the `ErrCodeEquals` helper function defined in `v2/awsv1shim/tfawserr` that uses the [`smithy.APIError`](https://pkg.go.dev/github.com/aws/smithy-go#APIError) interface.

```console
% go test ./...
ok  	github.com/hashicorp/aws-sdk-go-base/v2	5.656s
?   	github.com/hashicorp/aws-sdk-go-base/v2/internal/awsconfig	[no test files]
?   	github.com/hashicorp/aws-sdk-go-base/v2/internal/config	[no test files]
?   	github.com/hashicorp/aws-sdk-go-base/v2/internal/constants	[no test files]
ok  	github.com/hashicorp/aws-sdk-go-base/v2/internal/endpoints	0.313s
?   	github.com/hashicorp/aws-sdk-go-base/v2/internal/errs	[no test files]
ok  	github.com/hashicorp/aws-sdk-go-base/v2/internal/expand	0.629s
?   	github.com/hashicorp/aws-sdk-go-base/v2/internal/slices	[no test files]
?   	github.com/hashicorp/aws-sdk-go-base/v2/internal/test	[no test files]
?   	github.com/hashicorp/aws-sdk-go-base/v2/logging	[no test files]
?   	github.com/hashicorp/aws-sdk-go-base/v2/mockdata	[no test files]
?   	github.com/hashicorp/aws-sdk-go-base/v2/servicemocks	[no test files]
ok  	github.com/hashicorp/aws-sdk-go-base/v2/useragent	0.690s
```

### CODE REVIEW

**Review by YakDriver — APPROVED** [review]
LGTM 🎉

**Comment by ewbankkit** [comment]
Will wait for final OK from @gdavison on correct location.

File: tfawserr/awserr_test.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+// SPDX-License-Identifier: MPL-2.0
+
+package tfawserr
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/service/sts/types"
+	"github.com/aws/aws-sdk-go/aws"
```
Comment (ewbankkit) [inline tfawserr/awserr_test.go:11]: This should be `github.com/aws/aws-sdk-go-v2/aws`.

Comment (gdavison) [reply tfawserr/awserr_test.go:11]: Copy and paste :(

**Pull request merged**